bindings/node/lib/stateMachine.js

```diff
@@ -320,6 +320,9 @@ module.exports = function(modules) {
 
           if (request.bytesNeeded <= 0) {
             socket.end(resolve);
+            if (socket.unref) {
+              socket.unref();
```

Comment:
```suggestion
              socket.unref();
              rawSocket.unref();
```

for the proxy case?

Reply:
Handled the proxy case now with destroy.